components/rating/rating.js

```diff
@@ -0,0 +1,18 @@
+const ratingStars = document.querySelectorAll(".rating");
+
+const handleRating = (idx, ratingStars) => {
+  console.log("Hello");
+  // for chnaging the star color to blue to the preceeding stars to the clicked star
+  for (let i = 0; i <= idx; i++) {
+    if (!ratingStars[i].classList.contains('"rating-star-active"')) {
+      ratingStars[i].classList.add("rating-star-active");
+    }
+  }
```

Comment:
I have taken a different approach for this and on reading this code, I learned a new way to apply this.